embassy-stm32/src/eth/generic_smi.rs

```diff
@@ -62,6 +64,23 @@ impl GenericSMI {
 
 unsafe impl PHY for GenericSMI {
     fn phy_reset<S: StationManagement>(&mut self, sm: &mut S) {
+        // Detect SMI address
+        #[cfg(feature = "time")]
```

Comment:
it's surprising for the user that this is only available with `time`. Other drivers that need delays fall back to `cortex_m::asm::delay`. For example:

https://github.com/embassy-rs/embassy/blob/9a159a8db0638ea2971eed130a2ae34743da640a/embassy-stm32/src/adc/mod.rs#L83-L93

Maybe we should move that function somewhere shared, there's already a few copypastes of it around...

Reply:
This would be better for blocking code instead of dep. on `time`. For now I'll copypaste it, too)